datastream-server-restli/src/test/java/com/linkedin/datastream/server/TestCoordinator.java

```diff
@@ -2443,6 +2450,115 @@ public void testDatastreamDedupeMetadataCopy() throws Exception {
     Assert.assertEquals(datastreams[1].getMetadata().get(destMetaKey), destMetaVal);
   }
 
+  private class TestCoordinatorWithSpyZkAdapter extends Coordinator {
+
+    TestCoordinatorWithSpyZkAdapter(CachedDatastreamReader testDatastreamCache, Properties testConfig) throws DatastreamException {
+      super(testDatastreamCache, testConfig);
+    }
+
+    @Override
+    ZkAdapter createZkAdapter() {
+      return spy(new ZkAdapter(getConfig().getZkAddress(), getConfig().getCluster(),
+          getConfig().getDefaultTransportProviderName(), getConfig().getZkSessionTimeout(),
+          getConfig().getZkConnectionTimeout(), this));
+    }
+  }
+
+  @Test
+  public void testCoordinatorPerformPostBecomingLeaderTasks() throws Exception {
+    String testCluster = "testCoordinationSmoke3";
+    String testConnectorType = "testConnectorType";
+    String datastreamName1 = "datastream1";
+
+    Properties props = new Properties();
+    props.put(CoordinatorConfig.CONFIG_CLUSTER, testCluster);
+    props.put(CoordinatorConfig.CONFIG_ZK_ADDRESS, _zkConnectionString);
+    props.put(CoordinatorConfig.CONFIG_ZK_SESSION_TIMEOUT, String.valueOf(ZkClient.DEFAULT_SESSION_TIMEOUT));
+    props.put(CoordinatorConfig.CONFIG_ZK_CONNECTION_TIMEOUT, String.valueOf(ZkClient.DEFAULT_CONNECTION_TIMEOUT));
+
+    ZkClient zkClient = new ZkClient(_zkConnectionString);
+    _cachedDatastreamReader = new CachedDatastreamReader(zkClient, testCluster);
+    Coordinator instance1 = new TestCoordinatorWithSpyZkAdapter(_cachedDatastreamReader, props);
+    ZkAdapter spyZkAdapter1 = instance1.getZkAdapter();
+    instance1.addTransportProvider(DummyTransportProviderAdminFactory.PROVIDER_NAME,
+        new DummyTransportProviderAdminFactory().createTransportProviderAdmin(
+            DummyTransportProviderAdminFactory.PROVIDER_NAME, new Properties()));
+
+    TestHookConnector connector1 = new TestHookConnector("connector1", testConnectorType);
+    instance1.addConnector(testConnectorType, connector1, new BroadcastStrategy(Optional.empty()), false,
+        new SourceBasedDeduper(), null);
+    instance1.start();
+
+    String isLeaderMetricName = "Coordinator.isLeader";
+    Gauge<Integer> isLeader = DynamicMetricsManager.getInstance().getMetric(isLeaderMetricName);
+
+    Assert.assertEquals(isLeader.getValue().intValue(), 1);
+
+    //
+    // create datastream definitions under /testAssignmentBasic/datastream/datastream1
+    //
```

Comment:
nit: can we make the comment style consistent in this function? some comments have an extra // before and after, other's don't. And it is not exactly clear if this is only because of differentiating steps? If so, maybe number the steps instead to make it clear that those are step related comments?

Reply:
I have followed the comment style used in TestCoordinator file.

Reply:
Sure, if you think consistency is more important then leave it as is. In general though, this style of commenting is out of the ordinary from the rest of the code base